Gems/Terrain/Code/Tests/TerrainSystemBenchmarks.cpp

```diff
@@ -372,6 +384,56 @@ namespace UnitTest
         ->Args({ 4096, 1, static_cast<int>(AzFramework::Terrain::TerrainDataRequests::Sampler::EXACT) })
         ->Unit(::benchmark::kMillisecond);
 
+    BENCHMARK_DEFINE_F(TerrainSystemBenchmarkFixture, BM_ProcessHeightsRegionAsync)(benchmark::State& state)
+    {
+        // Run the benchmark
+        RunTerrainApiBenchmark(
+            state,
+            []([[maybe_unused]] float queryResolution, const AZ::Aabb& worldBounds,
+                AzFramework::Terrain::TerrainDataRequests::Sampler sampler)
+            {
+                auto perPositionCallback = []([[maybe_unused]] size_t xIndex, [[maybe_unused]] size_t yIndex, 
+                    const AzFramework::SurfaceData::SurfacePoint& surfacePoint, [[maybe_unused]] bool terrainExists)
+                {
+                    benchmark::DoNotOptimize(surfacePoint.m_position.GetZ());
+                };
+
+                AZStd::mutex completionMutex;
+                AZStd::condition_variable completionVariable;
+                AZStd::atomic_bool completionFlag = false;
+                auto completionCallback = [&completionMutex, &completionVariable, &completionFlag]([[maybe_unused]] AZStd::shared_ptr<AzFramework::Terrain::TerrainDataRequests::TerrainJobContext>)
+                {
+                    AZStd::unique_lock<AZStd::mutex> lock(completionMutex);
+                    completionFlag = true;
+                    completionVariable.notify_one();
+                };
+
```

Comment:
I think you can simplify this by just using a binary_semaphore that you acquire() in the main thread and release() in the completion callback.

Reply:
Quite possibly. This is left over from when I was spawning multiple async requests here and waiting for them all to complete (before I moved that logic into the implementation).

Reply:
Done, good suggestion.